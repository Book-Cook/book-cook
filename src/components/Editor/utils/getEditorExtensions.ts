// Import only the specific extensions we need from StarterKit
// instead of the entire StarterKit bundle
import { Document } from "@tiptap/extension-document";
import { Paragraph } from "@tiptap/extension-paragraph";
import { Text } from "@tiptap/extension-text";
import { Bold } from "@tiptap/extension-bold";
import { Italic } from "@tiptap/extension-italic";
import { Heading } from "@tiptap/extension-heading";
import { BulletList } from "@tiptap/extension-bullet-list";
import { OrderedList } from "@tiptap/extension-ordered-list";
import { ListItem } from "@tiptap/extension-list-item";
import { Link } from "@tiptap/extension-link";
import { Placeholder } from "@tiptap/extension-placeholder";
import { Underline } from "@tiptap/extension-underline";

export const getEditorExtensions = (placeholder: string) => [
  // Core document structure
  Document,
  Paragraph,
  Text,
  
  // Basic formatting
  Bold,
  Italic,
  Underline,
  
  // Headers (only levels we actually use)
  Heading.configure({ levels: [1, 2, 3] }),
  
  // Lists
  BulletList,
  OrderedList,
  ListItem,
  
  // Interactive features
  Link.configure({ openOnClick: false, autolink: true, linkOnPaste: true }),
  Placeholder.configure({ placeholder }),
];
