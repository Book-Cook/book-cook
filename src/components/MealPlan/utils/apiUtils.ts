/**
 * Reusable API utilities for meal plan operations
 */
import type { ApiPayload, MealMovePayload, MealReorderPayload } from "../types";

/**
 * Generic API call utility with error handling
 */
async function apiCall<T = any>(
  endpoint: string,
  method: 'GET' | 'POST' | 'PUT' | 'DELETE' = 'GET',
  payload?: Record<string, unknown>
): Promise<T> {
  const options: RequestInit = {
    method,
    headers: { 'Content-Type': 'application/json' },
  };

  if (payload && method !== 'GET') {
    options.body = JSON.stringify(payload);
  }

  const response = await fetch(endpoint, options);
  
  if (!response.ok) {
    throw new Error(`API call failed: ${response.statusText}`);
  }

  return response.json();
}

/**
 * Add meal to meal plan
 */
export function addMeal(payload: ApiPayload) {
  return apiCall('/api/meal-plans', 'POST', payload);
}

/**
 * Remove meal from meal plan
 */
export function removeMeal(date: string, time: string, mealIndex: number) {
  return apiCall(`/api/meal-plans/${date}/${time}/${mealIndex}`, 'DELETE');
}

/**
 * Move meal to different time slot
 */
export function moveMeal(payload: MealMovePayload) {
  const { date, time, mealIndex } = payload;
  return apiCall(`/api/meal-plans/${date}/${time}/move`, 'POST', payload);
}

/**
 * Reorder meals within same time slot
 */
export function reorderMeal(payload: MealReorderPayload) {
  const { date, time } = payload;
  return apiCall(`/api/meal-plans/${date}/${time}/reorder`, 'POST', payload);
}

/**
 * Batch operations for multiple meal changes
 */
export function batchMealOperations(operations: Array<{
  type: 'add' | 'remove' | 'move' | 'reorder';
  payload: Record<string, unknown>;
}>) {
  return apiCall('/api/meal-plans/batch', 'POST', { operations });
}