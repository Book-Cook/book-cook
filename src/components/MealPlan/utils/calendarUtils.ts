/**
 * Utility functions for calendar generation and date handling
 */

export const DAY_NAMES = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
export const DAY_NAMES_FULL = [
  "Sunday",
  "Monday", 
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday"
];

/**
 * Generate calendar days for a month view (6 weeks)
 */
export function getCalendarDays(currentDate: Date): Date[] {
  const year = currentDate.getFullYear();
  const month = currentDate.getMonth();
  
  // First day of the month
  const firstDay = new Date(year, month, 1);
  const startDate = new Date(firstDay);
  startDate.setDate(startDate.getDate() - firstDay.getDay());
  
  // Generate 6 weeks (42 days)
  const days: Date[] = [];
  const currentDay = new Date(startDate);
  
  for (let i = 0; i < 42; i++) {
    days.push(new Date(currentDay));
    currentDay.setDate(currentDay.getDate() + 1);
  }
  
  return days;
}

/**
 * Check if a date is today
 */
export function isToday(date: Date): boolean {
  const today = new Date();
  return date.toDateString() === today.toDateString();
}

/**
 * Check if a date is in the past
 */
export function isPast(date: Date): boolean {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return date < today;
}

/**
 * Check if a date is in the current month
 */
export function isCurrentMonth(date: Date, currentDate: Date): boolean {
  return date.getMonth() === currentDate.getMonth();
}

/**
 * Get the start and end dates of a week containing the given date
 */
export function getWeekRange(date: Date): { start: Date; end: Date } {
  const start = new Date(date);
  start.setDate(date.getDate() - date.getDay());
  start.setHours(0, 0, 0, 0);
  
  const end = new Date(start);
  end.setDate(start.getDate() + 6);
  end.setHours(23, 59, 59, 999);
  
  return { start, end };
}

/**
 * Get the start and end dates of a month
 */
export function getMonthRange(date: Date): { start: Date; end: Date } {
  const start = new Date(date.getFullYear(), date.getMonth(), 1);
  const end = new Date(date.getFullYear(), date.getMonth() + 1, 0);
  end.setHours(23, 59, 59, 999);
  
  return { start, end };
}

/**
 * Format date as YYYY-MM-DD for API calls
 */
export function formatDateForAPI(date: Date): string {
  return date.toISOString().split('T')[0];
}

/**
 * Navigate to previous/next period
 */
export function navigateDate(date: Date, direction: 'prev' | 'next', period: 'day' | 'week' | 'month'): Date {
  const newDate = new Date(date);
  
  switch (period) {
    case 'day':
      newDate.setDate(newDate.getDate() + (direction === 'next' ? 1 : -1));
      break;
    case 'week':
      newDate.setDate(newDate.getDate() + (direction === 'next' ? 7 : -7));
      break;
    case 'month':
      newDate.setMonth(newDate.getMonth() + (direction === 'next' ? 1 : -1));
      break;
  }
  
  return newDate;
}