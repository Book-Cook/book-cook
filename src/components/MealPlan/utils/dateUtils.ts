export const getWeekDates = (currentDate: Date): Date[] => {
  const dates: Date[] = [];
  const startOfWeek = new Date(currentDate);
  const day = startOfWeek.getDay();
  startOfWeek.setDate(startOfWeek.getDate() - day);
  
  for (let i = 0; i < 7; i++) {
    const date = new Date(startOfWeek);
    date.setDate(startOfWeek.getDate() + i);
    dates.push(date);
  }
  return dates;
};

export const isToday = (date: Date): boolean => {
  const today = new Date();
  return date.toDateString() === today.toDateString();
};

export const isPastDate = (date: Date): boolean => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  const compareDate = new Date(date);
  compareDate.setHours(0, 0, 0, 0);
  return compareDate < today;
};

export const formatDateString = (date: Date): string => {
  return date.toISOString().split("T")[0];
};

export const getCurrentTimePosition = (currentTime: Date, minHour: number, hourHeight: number): number | null => {
  const hours = currentTime.getHours();
  const minutes = currentTime.getMinutes();
  
  if (hours < minHour || hours > 22) {
    return null;
  }
  
  const totalMinutes = hours * 60 + minutes;
  const startMinutes = minHour * 60;
  return ((totalMinutes - startMinutes) / 60) * hourHeight;
};

export const getInitialScrollPosition = (minHour: number, hourHeight: number): number => {
  const now = new Date();
  const minutes = now.getHours() * 60 + now.getMinutes();
  const startMinutes = minHour * 60;
  const scrollPosition = ((minutes - startMinutes) / 60) * hourHeight;
  return Math.max(0, scrollPosition - 100);
};