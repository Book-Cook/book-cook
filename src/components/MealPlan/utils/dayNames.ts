/**
 * Day names constant
 */
export const dayNames = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];