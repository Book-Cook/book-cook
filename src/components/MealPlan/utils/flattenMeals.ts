/**
 * Flatten time slots into individual meal items with metadata
 */
import type { TimeSlot, FlattenedMeal } from "../types";

export const flattenMeals = (
  meals: TimeSlot[], 
  date: string
): FlattenedMeal[] => {
  const flattened: FlattenedMeal[] = [];
  
  meals.forEach(timeSlot => {
    timeSlot.meals.forEach((meal, index) => {
      flattened.push({
        id: `${date}-${timeSlot.time}-${index}`,
        meal,
        time: timeSlot.time,
        index,
      });
    });
  });
  
  return flattened;
};