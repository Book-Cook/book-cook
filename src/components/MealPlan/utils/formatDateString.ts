/**
 * Format a date to YYYY-MM-DD string
 */
export const formatDateString = (date: Date): string => {
  return date.toISOString().split("T")[0];
};