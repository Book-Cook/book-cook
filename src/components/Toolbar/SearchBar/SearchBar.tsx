import * as React from "react";
import { SearchBox } from "@fluentui/react-components";
import type {
  SearchBoxChangeEvent,
  InputOnChangeData,
} from "@fluentui/react-components";
import { useSearchBox } from "../../../context";
import { useRouter } from "next/router";

export const SearchBar = () => {
  const { searchBoxValue = "", onSearchBoxValueChange } = useSearchBox();

  const router = useRouter();
  const path = router.asPath;

  const onSearchBarChange = (
    _ev: SearchBoxChangeEvent,
    data: InputOnChangeData
  ) => {
    onSearchBoxValueChange(data.value);
  };

  const handleKeyDown = (event: any) => {
    if (event.key === "Enter") {
      if (path !== "/recipes") {
        // User is not on the home page, reroute
        router.push(`/recipes`);
      }
    }
  };

  return (
    <SearchBox
      value={searchBoxValue}
      onChange={onSearchBarChange}
      appearance="outline"
      placeholder="Search for snacks"
      onKeyDown={handleKeyDown}
    />
  );
};
