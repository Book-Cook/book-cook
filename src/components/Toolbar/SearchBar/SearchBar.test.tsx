import React from "react";
import { jest } from "@jest/globals";

import { SearchBar } from "./SearchBar";

import { render } from "../../../utils";

jest.mock("../../../context", () => ({
  useSearchBox: () => ({
    searchBoxValue: "",
    onSearchBoxValueChange: jest.fn(),
  }),
}));
jest.mock("next/router", () => jest.requireActual("next-router-mock"));

it("renders correctly and matches snapshot", () => {
  const { container } = render(<SearchBar />);
  expect(container).toMatchSnapshot();
});
