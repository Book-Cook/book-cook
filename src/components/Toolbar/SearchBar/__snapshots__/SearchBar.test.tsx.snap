// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <span
      class="fui-Input r1oeeo9n fui-SearchBox ___13yg2oi_15g8mz3 f1f5gg8d fhxju0i fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm fjuset5 f1xzfw5u"
    >
      <span
        class="fui-Input__contentBefore r1572tok fui-SearchBox__contentBefore"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.73 13.44a6.5 6.5 0 1 1 .7-.7l3.42 3.4a.5.5 0 0 1-.63.77l-.07-.06-3.42-3.41Zm-.71-.71A5.54 5.54 0 0 0 14 8.5a5.5 5.5 0 1 0-1.98 4.23Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <input
        class="fui-Input__input r12stul0 fui-SearchBox__input ___69a85s0_ca35df0 fk8j09s f1vdfbxk f18izjht fcoa6sg"
        placeholder="Search for snacks"
        type="search"
        value=""
      />
      <span
        class="fui-Input__contentAfter r1572tok fui-SearchBox__contentAfter ___yrbim60_1xrsupy f1cnd47f f1ufnopg fniina8 f3tsq5r f1a3p1vp"
      >
        <span
          aria-label="clear"
          class="fui-SearchBox__dismiss r1pvzcuu"
          role="button"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.09 4.22.06-.07a.5.5 0 0 1 .63-.06l.07.06L10 9.29l5.15-5.14a.5.5 0 0 1 .63-.06l.07.06c.18.17.2.44.06.63l-.06.07L10.71 10l5.14 5.15c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L10 10.71l-5.15 5.14a.5.5 0 0 1-.63.06l-.07-.06a.5.5 0 0 1-.06-.63l.06-.07L9.29 10 4.15 4.85a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</div>
`;
